src/common/config.go

```diff
@@ -196,23 +206,89 @@ func LoadDefaults(olPath string) error {
 		},
 	}
 
-	return checkConf()
+	return cfg, nil
 }
 
 // ParseConfig reads a file and tries to parse it as a JSON string to a Config
 // instance.
 func LoadConf(path string) error {
+	cfg, err := ReadInConf(path)
+	if err != nil {
+		return err
+	}
+
+	Conf = cfg
+	return checkConf()
+}
+
+func ReadInConf(path string) (*Config, error) {
 	configRaw, err := ioutil.ReadFile(path)
 	if err != nil {
-		return fmt.Errorf("could not open config (%v): %v", path, err.Error())
+		return nil, fmt.Errorf("could not open config (%v): %v", path, err.Error())
 	}
 
-	if err := json.Unmarshal(configRaw, &Conf); err != nil {
+	var templateConfig Config
+	if err := json.Unmarshal(configRaw, &templateConfig); err != nil {
 		fmt.Printf("Bad config file (%s):\n%s\n", path, string(configRaw))
-		return fmt.Errorf("could not parse config (%v): %v", path, err.Error())
+		return nil, fmt.Errorf("could not parse config (%v): %v", path, err.Error())
 	}
 
-	return checkConf()
+	return &templateConfig, nil
+}
+
+// this function returns the template confif struct with empty worker specific fields
```

Comment:
Why do we need this?  Let's not duplicate these defaults in two places (here, and GetDefaultWorkerConfig).